src/parser/shaman/restoration/modules/shadowlands/conduits/SwirlingCurrents.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import Events from 'parser/core/Events';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+
+import calculateEffectiveHealing from 'parser/core/calculateEffectiveHealing';
+
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import ItemHealingDone from 'interface/ItemHealingDone';
+
+class SwirlingCurrents extends Analyzer {
+
+  healingBoost = 0;
+
+  healing = 0;
+
+  targetsWithBoostedRiptides = [];
+
+  /**
+   * Whenever you cast a Gust of Mist procing ability it reduces the cooldown of Yu'lon or Chi-ji by .5 seconds as well as increasing their healing by x%
+   */
+  constructor(...args) {
+    super(...args);
+    this.active = true;
+
+    this.healingBoost = .2;//TODO Get from combat data when they EXPORT IT >:c
+
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell([SPELLS.HEALING_SURGE_RESTORATION, SPELLS.HEALING_WAVE, SPELLS.RIPTIDE]), this.normalizeBoost);
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.RIPTIDE), this.trackRiptide);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.RIPTIDE), this.removeRiptide);
+    this.addEventListener(Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.RIPTIDE), this.pandemicRiptide);
+  }
+
+  normalizeBoost(event){
+    if (this.selectedCombatant.hasBuff(SPELLS.SWIRLING_CURRENTS_BUFF.id) || this.targetsWithBoostedRiptides[event.targetID]) {
+      this.healing += calculateEffectiveHealing(event, this.healingBoost);
+    }
+  }
+
+  trackRiptide(event){
+    const targetID = event.targetID;
+    if (this.selectedCombatant.hasBuff(SPELLS.SWIRLING_CURRENTS_BUFF.id)) {
+      this.targetsWithBoostedRiptides[targetID]=targetID;
+    }else{
+      delete this.targetsWithBoostedRiptides[targetID];
+    }
+  }
+
+  removeRiptide(event){
+    delete this.targetsWithBoostedRiptides[event.targetID];
+  }
+
+  pandemicRiptide(event){
+    if (this.selectedCombatant.hasBuff(SPELLS.SWIRLING_CURRENTS_BUFF.id)) {
+     delete this.targetsWithBoostedRiptides[event.targetID];
```

Comment:
what does this actually do, when you cast a riptide on a target while you have the swirling buff, it removes it from the tracker? shouldn't that only happen when you don't have the buff

Reply:
i forgot a bang